app/tests/test_gs_mock.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from fastapi.testclient import TestClient
+from datetime import datetime
+from app.main import app
+
+client = TestClient(app)
+ver_prefix = "/api/v1"
+url = ver_prefix + "/gs/mock"
+
+
+def test_gs_mock_success():
+    payload = {
+        "start": "2023-01-01 00:00:00",
+        "end": "2023-01-01 01:00:00",
+        "delta_minutes": 60,
+    }
+    response = client.post(url, json=payload)
+    data = response.json()
+    assert response.status_code == 200
+    assert isinstance(data, list)
+    assert all(isinstance(item, bool) for item in data)
+
+
+def test_gs_mock_method_not_allowed():
+    # technically, we dont need to test this
+    # since fastapi will catch this :)
+    response = client.get(url)
+    assert response.status_code == 405
+
+
+def test_gs_mock_missing_json():
+    # technically, we dont need to test this
+    # since fastapi will catch this :)
+    response = client.post(url, json={})
+    assert response.status_code == 422
+
+
+def test_gs_mock_invalid_date_format():
+    payload = {
+        "start": "2023-01-01",
+        "end": "hello",
+        "delta_minutes": 15,
+    }
+    response = client.post(url, json=payload)
+    assert response.status_code == 422
+
+
+def test_gs_mock_alternative_date_format():
+    payload = {
```

Comment:
it even does the conversion on different date formats, as the validation is done by the `dattetime` class rather than me picking a random string